### PR TITLE
[Refactor]: Implement best practises

### PR DESCRIPTION
**Pull request by Smith-TI**
List of changes:

- [x] Add `.gitignore`
- [x] Add `.env` and remove all sensitive data from the code
- [x] Restructure folder structure to standard django app structure
- [x] Remove hardcoded values
- [x] Add/Improve comments and docstrings on functions
- [x] Add setup instructions in Readme
- [x] Add logs 
- [x] Add exception handling
- [x] Follow the DRY Principle
- [x] Remove unused code
- [x] Remove unused imports
- [x] Remove magic values
- [x] Implement Static typing
- [x] Remove `import *`
- [x] Integrate autopep8
- [x] Add missing dependencies in `requirements.txt`
- [x] Handle data leaks
- [x] Add String Representation of all models (`__str__` method)
- [x] Add Tests
- [x] Rename functions to more descriptive names
- [x] Add primary keys to models

### CODE REVIEW

File: restapi/utils.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+        else:
+            key = "{:02d}:00-{:02d}:15".format(hours, hours)
+
+        result.append([key, text])
+        print(key)
+
+    return result
+
+
+def multiThreadedReader(urls) -> list:
```
Comment (kushal-ti) [inline restapi/utils.py:98]: nitpick: use proper naming conventions for method names

Comment (kushal-ti) [inline restapi/utils.py:98]: nitpick: use proper naming conventions for function names

Comment (kushal-ti) [reply restapi/utils.py:98]: This particular method can be optimized by using proper concurrency primitives

File: restapi/views.py [hunk 2]
```diff
@@ -43,62 +57,54 @@ def balance(request):
...
-    queryset = User.objects.all()
-    serializer_class = UserSerializer
-    permission_classes = (AllowAny,)
+    '''View Set for a User'''
+    queryset: QuerySet = User.objects.all()
+    serializer_class: UserSerializer = UserSerializer
+    permission_classes: tuple = (AllowAny,)
 
 
 class category_view_set(ModelViewSet):
```
Comment (kushal-ti) [inline restapi/views.py:81]: nitpick: Use proper naming conventions in naming classes

File: restapi/views.py [hunk 3]
```diff
@@ -177,96 +197,36 @@ def get_queryset(self):
             expenses = Expenses.objects.filter(users__in=user.expenses.all())
         return expenses
 
+
 @api_view(['post'])
 @authentication_classes([])
 @permission_classes([])
-def logProcessor(request):
+def logProcessor(request) -> Response:
```
Comment (kushal-ti) [inline restapi/views.py:204]: nitpick: function name

File: restapi/views.py [hunk 4]
```diff
@@ -177,96 +197,36 @@ def get_queryset(self):
...
+    '''Handle POST method on processing log files'''
+    starttime = int(time.time() * 1000.0)
+
+    if not 'data' in request:
+        return Response(status=status.HTTP_400_BAD_REQUEST,
+                        data={"status": "failure", "reason": "field `data` missing in request"})
     data = request.data
     num_threads = data['parallelFileProcessingCount']
     log_files = data['logFiles']
+    logging.info(
```
Comment (kushal-ti) [inline restapi/views.py:214]: better to use a separate logger instance than the default logging module

**Review by kushal-ti — CHANGES_REQUESTED** [review]
Good attempt. Really impressed by how extensively static typing was implemented.
fix some of the nitpicks and i'll approve

File: restapi/utils.py (hunk 1 above)
Comment (Smith-TI) [reply restapi/utils.py]: > nitpick: use proper naming conventions for function names
Renamed to `multi_threaded_url_reader`

Comment (CapriciousRebel) [reply restapi/utils.py]: > This particular method can be optimized by using proper concurrency primitives

Replaced the `for loop` with `ThreadPoolExecutor` for concurrency and speed optimization

Comment (Smith-TI) [reply restapi/utils.py]: > This particular method can be optimized by using proper concurrency primitives

Replaced the `for loop` with `ThreadPoolExecutor` for concurrency and speed optimization

File: restapi/views.py (hunk 2 above)
Comment (Smith-TI) [reply restapi/views.py]: Renamed viewsets to PascalCase

File: restapi/views.py (hunk 3 above)
Comment (Smith-TI) [reply restapi/views.py]: Renamed `logProcessor` to `process_logs`

File: restapi/views.py (hunk 4 above)
Comment (Smith-TI) [reply restapi/views.py]: Added a custom logger `logger.py`

**Comment by kushal-ti** [comment]
looks good to me

**Pull request closed without merging**